src/Tests/Microsoft.NET.Build.Tests/GivenWeWantToRequireWindowsForDesktopApps.cs

```diff
@@ -0,0 +1,141 @@
+        [InlineData("UseWindowsForms")]
+        public void It_builds_on_windows_with_the_windows_desktop_sdk(string desktopProperty)
+        {
+            const string ProjectName = "WindowsDesktopSdkTest";
+
+            var asset = CreateWindowsDesktopSdkTestAsset(ProjectName, desktopProperty);
+            
+            var command = new BuildCommand(Log, Path.Combine(asset.Path, ProjectName));
+
+            command
+                .Execute("/restore")
+                .Should()
+                .Pass();
+        }
+
+        [PlatformSpecificTheory(Platform.Linux, Platform.Darwin, Platform.FreeBSD)]
+        [InlineData("UseWPF")]
+        [InlineData("UseWindowsForms")]
+        public void It_errors_on_nonwindows_with_the_windows_desktop_sdk(string desktopProperty)
+        {
+            const string ProjectName = "WindowsDesktopSdkErrorTest";
+
+            var asset = CreateWindowsDesktopSdkTestAsset(ProjectName, desktopProperty);
+            
+            var command = new BuildCommand(Log, Path.Combine(asset.Path, ProjectName));
+
+            command
+                .Execute("/restore")
+                .Should()
+                .Fail()
+                .And
+                .HaveStdOutContaining(Strings.WindowsDesktopFrameworkRequiresWindows);
+        }
+
+        [WindowsOnlyTheory]
+        [InlineData("Microsoft.WindowsDesktop.App")]
+        [InlineData("Microsoft.WindowsDesktop.App.WindowsForms")]
+        [InlineData("Microsoft.WindowsDesktop.App.WPF")]
+        public void It_builds_on_windows_with_a_framework_reference(string desktopFramework)
+        {
+            const string ProjectName = "WindowsDesktopReferenceTest";
+
+            var asset = CreateWindowsDesktopReferenceTestAsset(ProjectName, desktopFramework);
+            
+            var command = new BuildCommand(Log, Path.Combine(asset.Path, ProjectName));
+
+            command
+                .Execute("/restore")
+                .Should()
+                .Pass();
+        }
+
+        [PlatformSpecificTheory(Platform.Linux, Platform.Darwin, Platform.FreeBSD)]
+        [InlineData("Microsoft.WindowsDesktop.App")]
+        [InlineData("Microsoft.WindowsDesktop.App.WindowsForms")]
+        [InlineData("Microsoft.WindowsDesktop.App.WPF")]
+        public void It_errors_on_nonwindows_with_a_framework_reference(string desktopFramework)
+        {
+            const string ProjectName = "WindowsDesktopReferenceErrorTest";
+
+            var asset = CreateWindowsDesktopReferenceTestAsset(ProjectName, desktopFramework);
+            
+            var command = new BuildCommand(Log, Path.Combine(asset.Path, ProjectName));
+
+            command
+                .Execute("/restore")
+                .Should()
+                .Fail()
+                .And
+                .HaveStdOutContaining(Strings.WindowsDesktopFrameworkRequiresWindows);
+        }
+
+        private TestAsset CreateWindowsDesktopSdkTestAsset(string projectName, string desktopProperty)
```

Comment:
How about using `uiFramework` or something set to "WindowsForms" or "WPF" as the parameter?  I think that would be clearer than `desktopProperty`.

Reply:
Agreed.  I'll fix.